mondrian-generator.js

```diff
@@ -1,44 +1,56 @@
-var canvas = document.getElementById('compositionCanvas');
-var context = canvas.getContext('2d');
-
 function clearCanvas() {
   context.clearRect(0, 0, canvas.width, canvas.height);
 }
 
-/* Code courtesy of http://stackoverflow.com/questions/12796513/html5-canvas-to-png-file*/  
+function autoSaveImg(image) {
+  const link = document.createElement('a');
+  link.href = image;
+  link.download = 'new-mondrian.jpg';
+  document.body.appendChild(link);
+  link.click();
+  document.body.removeChild(link);
+}
+
+/* Code courtesy of http://stackoverflow.com/questions/12796513/html5-canvas-to-png-file*/
 function saveCanvas() {
+  const canvas = document.getElementById('compositionCanvas');
   var image = canvas.toDataURL("image/png");
   /* Change MIME type to trick the browser to downlaod the file instead of displaying it */
   image = image.replace(/^data:image\/[^;]*/, 'data:application/octet-stream');
   /* In addition to <a>'s "download" attribute, you can define HTTP-style headers */
   image = image.replace(/^data:application\/octet-stream/, 'data:application/octet-stream;headers=Content-Disposition%3A%20attachment%3B%20filename=mondrian.png');
   this.href = image;
+  autoSaveImg(image)
 }
 /* REGISTER DOWNLOAD HANDLER */
 /* Only convert the canvas to Data URL when the user clicks. 
    This saves RAM and CPU ressources in case this feature is not required. */
-document.getElementById('save').addEventListener('click', saveCanvas, false);
+// document.getElementById('save').addEventListener('click', saveCanvas, false);
+
+const colorToHexMap = {
+  red: '#fe0000',
+  yellow: '#ffff00',
+  blue: '#0000fe',
+  black: '#000000'
+}
+
+const colors = ['red', 'yellow', 'black', 'blue'];
 
-function composition() {
-  
-  colors = ['#c70000', '#f4b600', '#2d2bb4', 'black'];
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * (max - min + 1) + min);
+}
```

Comment:
`.getRandomInt` and other similar helper funcs removes need for lots of explanatory comments.